sysapi/sysapi/EncryptParam.c

```diff
@@ -93,9 +93,12 @@ TSS2_RC Tss2_Sys_SetEncryptParam(
             }
             else
             {
-                CopyMem( (uint8_t *)currEncryptParamBuffer, encryptParamBuffer,
-                        encryptParamSize,
-                        SYS_CONTEXT->tpmInBuffPtr + SYS_CONTEXT->maxCommandSize );
+                if (currEncryptParamBuffer + encryptParamSize >
+                        SYS_CONTEXT->tpmInBuffPtr + SYS_CONTEXT->maxCommandSize)
+                    return TSS2_SYS_RC_INSUFFICIENT_CONTEXT;
+
+               memmove((UINT8 *)currEncryptParamBuffer,
```

Comment:
You're casting away the `const` here. If I'm remembering memmove right this is the pointer to the location where the data is moved (the destination) so I'm wondering if the const is appropriate at all?

Reply:
This is because the Tss2_Sys_GetEncryptParam() expects const uint8_t ** so we have to cast somewhere. I'm not sure which place is better to cast memmove or Tss2_Sys_GetEncryptParam.
For now I will change the declaration to uint8_t *currEncryptParamBuffer